sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/ResponseBodyHandling.java

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.annotation.ServiceInterface;
+
+/**
+ * Enum that defines how to handle the body of an HTTP response.
+ */
+public enum ResponseBodyHandling {
+    /**
+     * Indicates that the body of an HTTP response shall be entirely ignored. No attempt to read the body will be made
+     * and the connection to the service will be closed as soon as a body-less {@link Response} instance is created.
+     */
+    IGNORE,
+
+    /**
+     * Indicates that the body of an HTTP response shall not be buffered into memory. The connection to the service will
+     * be kept alive to read the body at a future point in time when needed by the created {@link Response} instance.
+     */
+    STREAM,
+
+    /**
+     * Indicates that the body of an HTTP response shall be buffered into memory. The connection to the service will
+     * be closed after a {@link Response} instance containing the body is created. <b>This is the default behavior</b>.
+     */
+    BUFFER,
+
+    /**
+     * Indicates that the body of an HTTP response shall be buffered into memory and deserialized to the type specified
+     * in the {@link ServiceInterface} definition. The connection to the service will be closed after a {@link Response}
+     * instance containing the deserialized body is created.
```

Comment:
"deserialized to the type specified in the {@link ServiceInterface} definition" is a little too technical for the intended audience reading this JavaDoc. Maybe something like "shall be buffered into memory, and an attempt will be made to deserialize the body to a model type, accessible via the {@link Response#getValue} method" or similar.